### ISSUE TITLE
Memory leak during deserialization fiasco

### ISSUE THREAD

**Issue by Gricha** [ijl/orjson#89]
Hey all! Here is a minimal snippet I captured that will unboundedly leak memory:

```
import orjson

while 1:
    try:
        orjson.loads("Whatever")
    except orjson.JSONDecodeError:
        pass
```

I'm running that on Python 3.8.2 and with orjson 2.6.5. We've repro'd it both on Macs and Linux boxes.

**Comment by ijl** [comment]
It creates an orphan reference that is cleaned up when the garbage collector runs. Something about that loop prevents it? Fixed in 3.0.1.

**Comment by Gricha** [comment]
I confirmed that leaked in 2.6.5 by collecting gc every iteration, but I verified that it is fixed in 3.0.1. Thanks @ijl
